client/src/ui/atoms/Modal.tsx

```diff
@@ -12,59 +12,75 @@ interface Props {
-    isShowing,
-    hide,
-    children,
-    onAccept,
-    onCancel,
-    fullscreen = false,
-    buttonType = 'danger',
-    buttonText
-}: Props, ref: React.Ref<HTMLDivElement>): JSX.Element => {
-    const { t } = useTranslation();
-    const accept = (): void => {
-        if (onAccept) {
-            onAccept();
-        }
-        hide();
-    };
-    const cancel = (): void => {
-        if (onCancel) {
-            onCancel();
-        }
-        hide();
-    };    
-    return isShowing
-        ? ReactDOM.createPortal(
-              <React.Fragment>
-                  <div className="modal__overlay">
-                      <div className="modal__wrapper" aria-modal aria-hidden tabIndex={-1} role="dialog">
-                          <div className={`modal ${fullscreen ? 'modal__fullscreen' : ''}`} ref={ref}>
-                              <div className={`modal__content ${fullscreen ? 'modal__content--fullscreen' : ''}`}>
-                                  {children}
-                              </div>
-                              <div
-                                  className={`modal__buttons_container ${
-                                      fullscreen ? 'modal__buttons_container--fullscreen' : ''
-                                  }`}
-                              >
-                                  <div className={`modal__buttons ${fullscreen ? 'modal__buttons--fullscreen' : ''}`}>
-                                      <Button onClick={(): void => cancel()}>Cancel</Button>
-                                      <Button onClick={(): void => accept()} type={buttonType}>
-                                          {buttonText || t('ui:modal--default-button')}
-                                      </Button>
+const Modal = React.forwardRef(
+    (
+        {
+            isShowing,
+            hide,
+            children,
+            onAccept,
+            onCancel,
+            fullscreen = false,
+            buttonType = 'danger',
+            buttonText,
+            buttonDisabled = false,
+            bannerContent,
+        }: Props,
+        ref: React.Ref<HTMLDivElement>,
+    ): JSX.Element => {
+        const { t } = useTranslation();
+        const accept = (): void => {
+            if (onAccept) {
+                onAccept();
+            }
+            hide();
+        };
+        const cancel = (): void => {
+            if (onCancel) {
+                onCancel();
+            }
+            hide();
+        };
+        return isShowing
+            ? ReactDOM.createPortal(
+                  <React.Fragment>
+                      <div className="modal__overlay">
+                          <div className="modal__wrapper" aria-modal aria-hidden tabIndex={-1} role="dialog">
+                              <div className={`modal ${fullscreen ? 'modal__fullscreen' : ''}`} ref={ref}>
+                                  {fullscreen && bannerContent && <Banner>{bannerContent}</Banner>}
```

Comment:
I"m not sure this should be a feature of the `Modal` but rather part of the modals `children` content.

Reply:
The modal content is used to enforce the 66% width so the banner doesn't look correct if we put it there.

Reply:
Could we apply a `100vw` to the banner instead of `100%`? The banner seems pretty bespoke to the PeoplePicker so not sure it makes sense for it to be a proped option. Or could we do the centring 66% with a children component of `Modal` rather than setting it by default?

Reply:
It looks like we could do with a common `CenteredContent` component as the modal content, form content and dashboard share similar positioning

Reply:
I think id rather make it a class we apply than a component, having en extra element in the dom tree for it seems overkill.